### PR TITLE
(core) convert applicationReader to TS

### PR DESCRIPTION
**Pull request by anotherchrisberry**

### CODE REVIEW

**Comment by anotherchrisberry** [comment]
@danielpeach PTAL (Zan and Adam are both on vacation right now)

File: app/scripts/modules/core/application/service/applications.read.service.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    this.setDisabledDataSources(application);
+  }
+
+  private setDisabledDataSources(application: Application) {
+    const allDataSources: ApplicationDataSource[] = application.dataSources,
+          appDataSources: IApplicationDataSourceAttribute = application.attributes.dataSources;
+    if (!appDataSources) {
+      allDataSources.filter(ds => ds.optIn).forEach(ds => this.disableDataSource(ds, application));
+    } else {
+      allDataSources.forEach((ds: any) => {
```
Comment (danielpeach) [inline app/scripts/modules/core/application/service/applications.read.service.ts]: Seems like the type should either be implicit or `ApplicationDataSource` here.

File: app/scripts/modules/core/application/service/applications.read.service.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        if (ds.optional) {
+          if (ds.optIn && !appDataSources.enabled.includes(ds.key)) {
+            this.disableDataSource(ds, application);
+          }
+          if (!ds.optIn && ds.optional && appDataSources.disabled.includes(ds.key)) {
```
Comment (danielpeach) [inline app/scripts/modules/core/application/service/applications.read.service.ts]: I think `ds.optional` is redundant here given the check on line 59.

Comment (anotherchrisberry) [inline app/scripts/modules/core/application/service/applications.read.service.ts]: Nice catch, will remove

File: app/scripts/modules/core/application/service/applications.read.service.ts (hunk 1 above)
Comment (anotherchrisberry) [inline app/scripts/modules/core/application/service/applications.read.service.ts]: Yep, good catch. Will fix.

**Pull request merged**